geode-assembly/src/test/java/org/apache/geode/management/internal/cli/commands/LogsAreFullyRedactedAcceptanceTest.java

```diff
@@ -0,0 +1,135 @@
+
+package org.apache.geode.management.internal.cli.commands;
+
+import static org.apache.geode.distributed.ConfigurationProperties.LOG_LEVEL;
+import static org.apache.geode.distributed.ConfigurationProperties.SECURITY_MANAGER;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Properties;
+import java.util.Scanner;
+
+import org.apache.commons.io.FileUtils;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.examples.SimpleSecurityManager;
+import org.apache.geode.management.internal.cli.util.CommandStringBuilder;
+import org.apache.geode.test.junit.categories.AcceptanceTest;
+import org.apache.geode.test.junit.rules.gfsh.GfshRule;
+
+@Category(AcceptanceTest.class)
+public class LogsAreFullyRedactedAcceptanceTest {
+
+  private static String PASSWORD = "abcdefghijklmn";
+
+  // Don't put "password" in all the key names, since that is one of our taboo words.
+  private static final String gfshSysEnvUsername = "log-no-test.security-username";
+  private static final String gfshSysEnvUsernameValue = "test.sysenv-username";
+  private static final String gfshSysEnvPassword = "log-no-test.security-password";
+  private static final String gfshSysEnvPasswordValue = PASSWORD + "-sysenv";
+
+  private static final String gfshSysEnvUsernamePost = "security-username.log-no-test";
+  private static final String gfshSysEnvUsernameValuePost = "sysenv-username.test";
+  private static final String gfshSysEnvPasswordPost = "security-password.log-no-test";
+  private static final String gfshSysEnvPasswordValuePost = PASSWORD + "-post";
+
+
+
+  private File propertyFile;
+  private File securityPropertyFile;
+
+  @Rule
+  public GfshRule gfsh = new GfshRule();
+
+  @BeforeClass
+  public static void addSecuritySystemProperties() {
+    System.setProperty(gfshSysEnvUsername, gfshSysEnvUsernameValue);
+    System.setProperty(gfshSysEnvPassword, gfshSysEnvPasswordValue);
+    System.setProperty(gfshSysEnvUsernamePost, gfshSysEnvUsernameValuePost);
+    System.setProperty(gfshSysEnvPasswordPost, gfshSysEnvPasswordValuePost);
+  }
+
+  @Before
+  public void createPropertyFiles() throws IOException {
+    propertyFile = gfsh.getTemporaryFolder().newFile("geode.properties");
+    securityPropertyFile = gfsh.getTemporaryFolder().newFile("security.properties");
+
+    Properties properties = new Properties();
+    properties.setProperty(LOG_LEVEL, "debug");
+    try (FileOutputStream fileOutputStream = new FileOutputStream(propertyFile)) {
+      properties.store(fileOutputStream, null);
+    }
+
+    Properties securityProperties = new Properties();
+    securityProperties.setProperty(SECURITY_MANAGER, SimpleSecurityManager.class.getName());
```

Comment:
I would still use a custom security manager that would accept that special password, and scan the logs only for the occurrence of that word.